### PR TITLE
Role-based Authorization Handler

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR implements the Role-based Authorization handler.  It includes an update to the `RoleBaseAuthorizationStore` to provide a function to return the roles assigned to a given identity.

### CODE REVIEW

**Review by ltseeley — APPROVED** [review]

**Review by ltseeley — DISMISSED** [review]

File: splinterd/Cargo.toml [hunk 1]
```diff
@@ -81,6 +81,7 @@ experimental = [
     # The following features are experimental:
     "auth",
     "authorization",
+    "authorization-handler-rbac",
```
Comment (ltseeley) [inline splinterd/Cargo.toml:84]: Can we shorten this to `auth-handler-rbac` for brevity? The maintenance and allow_keys ones would especially benefit from this (they'll match). Can you also guard the handler itself in libsplinter?

Comment (peterschwarz) [reply splinterd/Cargo.toml:84]: I think the name is correct, but I have added the guard to libsplinter as well.

**Review by vaporos — APPROVED** [review]

**Review by ltseeley — APPROVED** [review]

**Review by vaporos — APPROVED** [review]
